src/librustc_mir/hair/pattern/mod.rs

```diff
@@ -403,9 +414,11 @@ impl<'a, 'tcx> PatternContext<'a, 'tcx> {
             PatKind::Lit(ref value) => self.lower_lit(value),
 
             PatKind::Range(ref lo_expr, ref hi_expr, end) => {
-                match (self.lower_lit(lo_expr), self.lower_lit(hi_expr)) {
-                    (PatternKind::Constant { value: lo },
-                     PatternKind::Constant { value: hi }) => {
+                match (
```

Comment:
To avoid miscompilations down the line, could you also replace the `_ => PatternKind::Wild` line with this?

```Rust
                    ref pats => {
                        self.tcx.delay_span_bug(
                            pat.span, format!("found bad range pattern `{:?}` \
                                               outside of error recovery", pats));

                        PatternKind::Wild
                    }
```

Reply:
Changed this.